vespa-athenz/src/main/java/com/yahoo/vespa/athenz/tls/AthenzX509CertificateUtils.java

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.athenz.tls;
+
+import com.yahoo.vespa.athenz.api.AthenzIdentity;
+import com.yahoo.vespa.athenz.api.AthenzRole;
+import com.yahoo.vespa.athenz.utils.AthenzIdentities;
+
+import java.security.cert.X509Certificate;
+import java.util.List;
+
+import static com.yahoo.vespa.athenz.tls.SubjectAlternativeName.Type.RFC822_NAME;
+
+/**
+ * Utility methods for Athenz issued x509 certificates
+ *
+ * @author bjorncs
+ */
+public class AthenzX509CertificateUtils {
+
+    private static final String COMMON_NAME_ROLE_DELIMITER = ":role.";
+
+    private AthenzX509CertificateUtils() {}
+
+    public static boolean isAthenzRoleCertificate(X509Certificate certificate) {
+        return isAthenzIssuedCertificate(certificate) &&
+                X509CertificateUtils.getSubjectCommonNames(certificate).get(0).contains(COMMON_NAME_ROLE_DELIMITER);
+    }
+
+    public static boolean isAthenzIssuedCertificate(X509Certificate certificate) {
```

Comment:
Consider removing this check (assuming trust store is set up to only trust these issuers)